platforms/software/dependency-management/src/main/java/org/gradle/internal/artifacts/configurations/AbstractRoleBasedConfigurationCreationRequest.java

```diff
@@ -64,14 +72,26 @@ public Configuration verifyExistingConfigurationUsage(Configuration conf) {
         ConfigurationInternal internalConf = (ConfigurationInternal) conf;
 
         if (!getRole().isUsageConsistentWithRole(conf)) {
-            warnAboutNeedToMutateUsage(internalConf);
-            if (internalConf.usageCanBeMutated()) {
-                internalConf.setAllowedUsageFromRole(getRole());
-            } else {
-                failOnInabilityToMutateUsage();
-            }
+            failOnNeedToMutateUsage(internalConf);
```

Comment:
There are a number of warnings in `verifyExistingConfigurationUsage` and I think it is likely worth addressing them all at once.

At the beginning of this method, we call `warnAboutReservedName`. This method _unconditionally_ emits a deprecation warning. 

That means that as of 9.0 `warnAboutReservedName` becomes `failDueToReservedName`, and any code that occurs after this method call can effectively just be deleted.

I think trying to change `warnAboutNeedToMutateUsage` before `warnAboutReservedName` is addressed is probably pointless -- as it is effectively just dead code

Reply:
After the recent pain of trying to address other related deprecations together, I'm going to do Configuration-related changes strictly one at a time for my own sanity.

This just needs a failing sample addressed + the change you identified above.  I'll get this finished and in the merge queue and then do that immediately afterwards stacked on top of this change.